pinot-segment-local/src/main/java/org/apache/pinot/segment/local/data/manager/TableDataManagerConfig.java

```diff
@@ -18,119 +18,75 @@
+    if (peerSegmentDownloadScheme != null) {
+      return peerSegmentDownloadScheme;
+    }
+    return _instanceDataManagerConfig.getSegmentPeerDownloadScheme();
   }
 
-  public static TableDataManagerConfig getDefaultHelixTableDataManagerConfig(
-      InstanceDataManagerConfig instanceDataManagerConfig, String tableNameWithType) {
-    Configuration defaultConfig = new PropertiesConfiguration();
-    defaultConfig.addProperty(TABLE_DATA_MANAGER_NAME, tableNameWithType);
-    defaultConfig.addProperty(TABLE_DATA_MANAGER_DATA_DIRECTORY,
-        instanceDataManagerConfig.getInstanceDataDir() + "/" + tableNameWithType);
-    defaultConfig.addProperty(TABLE_DATA_MANAGER_CONSUMER_DIRECTORY, instanceDataManagerConfig.getConsumerDir());
-    TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableNameWithType);
-    Preconditions.checkNotNull(tableType);
-    defaultConfig.addProperty(TABLE_DATA_MANAGER_TYPE, tableType.name());
-    defaultConfig.addProperty(TABLE_DELETED_SEGMENTS_CACHE_SIZE,
-        instanceDataManagerConfig.getDeletedSegmentsCacheSize());
-    defaultConfig.addProperty(TABLE_DELETED_SEGMENTS_CACHE_TTL_MINUTES,
-        instanceDataManagerConfig.getDeletedSegmentsCacheTtlMinutes());
-    // allow null
-    String segmentPeerDownloadScheme = instanceDataManagerConfig.getSegmentPeerDownloadScheme();
-    LOGGER.info("instance level segment peer download scheme = {}", segmentPeerDownloadScheme);
-    defaultConfig.addProperty(TABLE_PEER_DOWNLOAD_SCHEME, segmentPeerDownloadScheme);
-
-
-    // copy auth-related configs
-    instanceDataManagerConfig.getConfig().subset(TABLE_DATA_MANAGER_AUTH).toMap()
-        .forEach((key, value) -> defaultConfig.setProperty(TABLE_DATA_MANAGER_AUTH + "." + key, value));
-
-    return new TableDataManagerConfig(defaultConfig);
+  public int getTableDeletedSegmentsCacheSize() {
+    return _instanceDataManagerConfig.getDeletedSegmentsCacheSize();
   }
 
-  public void overrideConfigs(TableConfig tableConfig) {
-    // Override table level configs
-
-    _tableDataManagerConfig.addProperty(TABLE_IS_DIMENSION, tableConfig.isDimTable());
-    SegmentsValidationAndRetentionConfig segmentConfig = tableConfig.getValidationConfig();
-    if (segmentConfig != null && segmentConfig.getPeerSegmentDownloadScheme() != null) {
-      _tableDataManagerConfig.setProperty(TABLE_PEER_DOWNLOAD_SCHEME, segmentConfig.getPeerSegmentDownloadScheme());
-    }
-    LOGGER.info("final segment peer download scheme = {}", getTablePeerDownloadScheme());
+  public int getTableDeletedSegmentsCacheTtlMinutes() {
+    return _instanceDataManagerConfig.getDeletedSegmentsCacheTtlMinutes();
+  }
 
-    // If we wish to override some table level configs using table config, override them here
-    // Note: the configs in TableDataManagerConfig is immutable once the table is created, which mean it will not pick
-    // up the latest table config
+  public Configuration getAuthConfig() {
```

Comment:
Is the auth config same for the server and the table data manager ?

Reply:
Based on the current implementation, it is the same. I assume we will need table level auth in the future